server/src/configs/validators.rs

```diff
@@ -77,8 +77,8 @@ impl Validatable<ServerError> for SegmentConfig {
 impl Validatable<ServerError> for MessageSaverConfig {
     fn validate(&self) -> Result<(), ServerError> {
         if self.interval == 0 {
-            // TODO: use a custom error
-            panic!("Message saver interval must be greater than 0.")
+            error!("Saver Interval size cannot be zero, it must be greater than 0.");
```

Comment:
can you return error with String, like in other impl Validatable here? eg.
![obraz](https://github.com/iggy-rs/iggy/assets/5490304/f8ee9432-e9c6-42e3-a196-90a957bba08e)

Reply:
Do you want me to change the signature for InvalidConfiguration enum to InvalidConfiguration(String). This would effect changing code in few other places as well. 

```
Previously

#[derive(Debug, Error)]
pub enum ServerError {
    #[error("IO error")]
    IoError(#[from] io::Error),
    #[error("Connection error")]
    ConnectionError(#[from] ConnectionError),
    #[error("Invalid configuration provider: {0}")]
    InvalidConfigurationProvider(String),
    #[error("Cannot load configuration: {0}")]
    CannotLoadConfiguration(String),
    #[error("Invalid configuration")]
    InvalidConfiguration(String),

```


```
Now
#[derive(Debug, Error)]
pub enum ServerError {
    #[error("IO error")]
    IoError(#[from] io::Error),
    #[error("Connection error")]
    ConnectionError(#[from] ConnectionError),
    #[error("Invalid configuration provider: {0}")]
    InvalidConfigurationProvider(String),
    #[error("Cannot load configuration: {0}")]
    CannotLoadConfiguration(String),
    #[error("Invalid configuration")]
    InvalidConfiguration,
```

Reply:
Ok, you can leave it as is. We are still growing fast, so no point in being so picky regarding errors - we don't need that kind of granularity (at least now).